control/server.py

```diff
@@ -59,9 +60,37 @@ class GatewayServer:
         discovery_pid: Subprocess running Ceph nvmeof discovery service
     """
 
+    class RPCGuard:
```

Comment:
Apparently we don't need to implement Guard design pattern, see here 

### Using locks in the with statement
All of the objects provided by a module that has acquire() and release() methods can be used as context managers for a with statement. The acquire() method will be called when the block is entered, and release() will be called when the block is exited

(see https://docs.python.org/3/library/threading.html#with-locks)

with some_lock:
    # do something...

Reply:
Well, we do not need to reimplement a wheel, I think.

Reply:
We didn't. I sued well knows python objects. This entire class is a few lines long and it allows me to use the locks in the exact way I want.